src/background/user-activity.ts

```diff
@@ -27,12 +28,16 @@ export function scheduleAlarms() {
   });
 }
 
+async function getIdleTimeout() {
+  const preferences = await globalPreferences.getPreferences();
+  return preferences.autoLockTimeout;
+}
+
 export async function handleAlarm(alarm: browser.Alarms.Alarm) {
   if (alarm.name === 'lastActiveCheck') {
     const lastActive = await getLastActive();
-    // const ONE_DAY = 1000 * 60 * 60 * 24;
-    const HALF_A_DAY = 1000 * 60 * 60 * 12;
-    if (lastActive && Date.now() - lastActive > HALF_A_DAY) {
+    const idleTimeout = await getIdleTimeout();
+    if (idleTimeout && lastActive && Date.now() - lastActive > idleTimeout) {
```

Comment:
can `idleTimeout` be falsy? If not, I'd compare explicitly to zero
If yes, then I'd add a fallback to logout after some predefined time

Reply:
I implemented 0 as 'none'.
We can change this and make this option more obvious